internal/api/external.go

```diff
@@ -587,7 +592,7 @@ func getErrorQueryString(err error, errorID string, log logrus.FieldLogger) *url
 		q.Set("error_description", e.Description)
 		log.WithError(e.Cause()).Info(e.Error())
 	case ErrorCause:
-		return getErrorQueryString(e.Cause(), errorID, log)
+		return getErrorQueryString(e.Cause(), errorID, log, q)
```

Comment:
Is this recursive? Why is this necessary to be recursive?

Reply:
yeah it came as a recursive function - not sure why it was designed as such

Reply:
this is fine though right? it's a tail recursion so no perf implications

Reply:
Going to mark as resolve as it shouldn't cause any significant performance decrease. If there are readability or other concerns let me know though and we can relook